framework/src/utils/JsonSyntaxTree.C

```diff
@@ -120,22 +124,30 @@ JsonSyntaxTree::addParameters(const std::string & parent, const std::string & pa
   }
   if (_search != "" && count == 0)
     // no parameters that matched the search string
-    return;
+    return false;
 
   moosecontrib::Json::Value & json = getJson(parent, path, is_type);
+
   if (is_action)
   {
     json[action]["parameters"] = all_params;
-    json[action]["tasks"].append(task_name);
+    json[action]["description"] = params->getClassDescription();
     json[action]["action_path"] = path;
+    auto & fi = json[action]["file_info"];
```

Comment:
Bracket access on RHS: This is our infamous "accidental insertion" pattern for std::map. Is this access OK here if the json object is empty? If not can you add a MOOSE assert or something?